vcd/datasource_vcd_network_test.go

```diff
@@ -0,0 +1,367 @@
+// the previous results.
+func getAvailableNetworks() error {
+	if networksSearched {
+		return nil
+	}
+	vcdClient, err := getTestVCDFromJson(testConfig)
+	if err != nil {
+		return fmt.Errorf("error getting client configuration: %s", err)
+	}
+	err = vcdClient.Authenticate(testConfig.Provider.User, testConfig.Provider.Password, testConfig.Provider.SysOrg)
+	if err != nil {
+		return fmt.Errorf("authentication error: %s", err)
+	}
+	org, err := vcdClient.GetAdminOrgByName(testConfig.VCD.Org)
+	if err != nil {
+		return fmt.Errorf("org not found : %s", err)
+	}
+	vdc, err := org.GetVDCByName(testConfig.VCD.Vdc, false)
+	if err != nil {
+		return fmt.Errorf("vdc not found : %s", err)
+	}
+
+	for _, net1 := range vdc.Vdc.AvailableNetworks {
+		for _, net2 := range net1.Network {
+			network, err := vdc.GetOrgVdcNetworkByName(net2.Name, false)
+			if err == nil {
+				client := VCDClient{
+					VCDClient:       vcdClient,
+					SysOrg:          testConfig.Provider.SysOrg,
+					Org:             testConfig.VCD.Org,
+					Vdc:             testConfig.VCD.Vdc,
+					MaxRetryTimeout: testConfig.Provider.MaxRetryTimeout,
+					InsecureFlag:    true,
+				}
+				networkType, parent := detectNetwork(&client, vdc, network)
+				_, ok := availableNetworks[networkType]
+				if !ok {
+					if networkType == "vcd_network_isolated" {
+						// Make sure the IPScope structure is reachable for isolated networks
+						if network.OrgVDCNetwork.Configuration != nil &&
+							network.OrgVDCNetwork.Configuration.IPScopes != nil &&
+							len(network.OrgVDCNetwork.Configuration.IPScopes.IPScope) > 0 {
+							availableNetworks[networkType] = networkRec{network.OrgVDCNetwork, parent}
+						}
+					} else {
+						availableNetworks[networkType] = networkRec{network.OrgVDCNetwork, parent}
+					}
+				}
+			}
+		}
+	}
+
+	networksSearched = true
+	return nil
+}
+
+func TestAccVcdNetworkDirectDS(t *testing.T) {
+	// This test requires access to the vCD before filling templates
+	// Thus it won't run in the short test
+	if vcdShortTest {
+		t.Skip(acceptanceTestsSkipped)
+		return
+	}
+
+	err := getAvailableNetworks()
+	if err != nil {
+		fmt.Printf("%s\n", err)
+		t.Skip("error getting available networks")
+		return
+	}
+	if len(availableNetworks) == 0 {
+		t.Skip("No networks found - data source test skipped")
+		return
+	}
+
+	networkType := "vcd_network_direct"
+	data, ok := availableNetworks[networkType]
+	if !ok {
+		t.Skip("no direct network found ")
+		return
+	}
+
+	template := networkTemplates[networkType]
+	var params = StringMap{
+		"Org":             testConfig.VCD.Org,
+		"VDC":             testConfig.VCD.Vdc,
+		"ExternalNetwork": data.parent,
+		"NetworkName":     data.network.Name,
+		"FuncName":        "TestNetworkDirectDS",
+		"Tags":            "network",
+	}
+	configText := templateFill(template, params)
+	debugPrintf("#[DEBUG] CONFIGURATION: %s", configText)
+
+	resource.Test(t, resource.TestCase{
```

Comment:
Worth trying out `resource.ParallelTest` to shave of some test time.

Reply:
this one can't be parallelized without changing the IP ranges.